src/adaptors/amqp/qd_connector.c

```diff
@@ -504,11 +485,14 @@ void qd_connector_remove_connection(qd_connector_t *connector, bool final, const
     qd_connection_t *ctx = connector->qd_conn;
     if (!connector->is_data_connector && !connector->oper_status_down  && !final) {
         connector->oper_status_down = true;
-        vflow_set_string(connector->vflow_record, VFLOW_ATTRIBUTE_OPER_STATUS, "down");
-        vflow_inc_counter(connector->vflow_record, VFLOW_ATTRIBUTE_DOWN_COUNT, 1);
-        vflow_set_timestamp_now(connector->vflow_record, VFLOW_ATTRIBUTE_DOWN_TIMESTAMP);
-        vflow_set_string(connector->vflow_record, VFLOW_ATTRIBUTE_RESULT, condition_name        ? condition_name : "unknown");
-        vflow_set_string(connector->vflow_record, VFLOW_ATTRIBUTE_REASON, condition_description ? condition_description : "");
+        qd_connector_config_t *ctor_config = connector->ctor_config;
```

Comment:
Is it possible that the loss of the "old" control connection might be registered here and set the OPER_STATUS to "down" even though the "new" control connection is up?

Reply:
BTW, I am writing a test for this as well.

Reply:
This seems like a clever solution.  The only question I have is whether this breaks when the link doesn't use TLS (i.e. there's no SslProfile).  However, I don't think a connector would ever be rotated if there wasn't an SslProfile, so this is probably fine.